tests/AgentTests/FunctionalAPITests.hs

```diff
@@ -127,6 +130,50 @@ functionalAPITests t = do
       testServerMatrix2 t testSwitchAsync
     describe "should delete connection during rotation" $
       testServerMatrix2 t testSwitchDelete
+  describe "SMP basic auth" $ do
+    describe "with server auth" $ do
+      --                                       allow NEW | server auth, v | clnt1 auth, v  | clnt2 auth, v    |  2 - success, 1 - JOIN fail, 0 - NEW fail
+      it "success                " $ testBasicAuth t True (Just "abcd", 5) (Just "abcd", 5) (Just "abcd", 5) `shouldReturn` 2
+      it "disabled               " $ testBasicAuth t False (Just "abcd", 5) (Just "abcd", 5) (Just "abcd", 5) `shouldReturn` 0
+      it "NEW fail, no auth      " $ testBasicAuth t True (Just "abcd", 5) (Nothing, 5) (Just "abcd", 5) `shouldReturn` 0
+      it "NEW fail, no auth both " $ testBasicAuth t True (Just "abcd", 5) (Nothing, 5) (Nothing, 5) `shouldReturn` 0
```

Comment:
not sure what's the point in distinguishing these test cases when `testBasicAuth` + `testCreateQueueAuth` are written in a way that makes the first impossible to succeed..